modules/aws/lb/main.tf

```diff
@@ -0,0 +1,301 @@
+###########################
+# Provider Configuration
+###########################
+terraform {
+  required_version = ">= 1.0.0"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
+    }
+  }
+}
+
+###########################
+# Locals
+###########################
+
+locals {
+  is_application = var.load_balancer_type == "application"
+  is_network     = var.load_balancer_type == "network"
+}
+
+###########################################################
+# Load Balancer
+###########################################################
+
+resource "aws_lb" "load_balancer" {
+  # Common settings
+  name                       = var.name
```

Comment:
Consider whether we want to use `name` or `name_prefix`. My rule of thumb has been that if there could be multiples of something I default to using `name_prefix` as they avoid collisions at the outset. Rare that we care if the name includes the datetime.

Reply:
I would like to talk more about this. I think when it comes to global name spaces it makes sense, but maybe we can make it optional here?

just initial thoughts. going to come back to this.

Reply:
Ah that's fair, I had assumed it was global namespace but if it's just `tag { Name = XYZ }` then you're right.

Reply:
<img width="1130" alt="image" src="https://github.com/user-attachments/assets/9c907c2b-edfc-465f-9280-4a48d2b112ea" />
Looks like unique within your account so we should use name_prefix to just alleviate collisions due to user mistake.

Reply:
updated